src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -287,6 +287,11 @@ public String run(Path appPath, String appName) throws IOException, InterruptedE
      */
     @Override
     public boolean runUntilEnd() throws IOException, InterruptedException {
+        Path app = Objects.requireNonNull(paths.getAppPath(), "Application path can't be null")
+                .resolve(Objects.requireNonNull(projectConfiguration.getAppName(), "Application name can't be null"));
+        if (!Files.exists(app)) {
+            throw new IOException("Application not found at path " + app.toString());
+        }
         Process runProcess = startAppProcess(paths.getAppPath(), projectConfiguration.getAppName());
```

Comment:
Although, there is inherently nothing wrong with this code, the intent would be clearer if these tests have their own lines.

```
final Path appPath = Objects.requireNonNull(paths.getAppPath(), "Application path can't be null");
final String appName = Objects.requireNonNull(projectConfiguration.getAppName(), "Application name can't be null");
if (!Files.exists(appPath.resolve(appName))) {
    throw new IOException("Application not found at path " + appName);
}
Process runProcess = startAppProcess(appPath, appName);
```

Reply:
changed